src/auth/types.ts

```diff
@@ -4,8 +4,34 @@ export interface User {
   email: string;
   isAuthorizer: boolean;
   isTreasurer: boolean;
-  bsb?: string;
-  accountNum?: string;
   photoURL: string;
   token?: string;
+
+  // these are set in bank form
+  bsb?: string;
+  accountNum?: string;
+  // TODO: add an account name
 }
+
+// creates an user instance
+export function createUser(
+  obj: any
+): User {
+  if (obj == null){
```

Comment:
Might not be a big issue depending on how your logic is setup but shallow equality here means undefined obj also passes the check (i.e when obj is undefined).

Reply:
Probably keep it as shallow compare; we need undefined to be truthy else the destructuring below fails.